Coffee Editor NG

eneufeld opened the pull request:
This is a refactoring of the whole web page.

planger committed:
Improve front page

planger committed:
Udpate container

planger committed:
Add initial structure for documentation

planger committed:
Add Eugens feedback

eneufeld committed:
add tree based editor documentation

martin-fleck-at committed:
Documentation: Language Definition and Langium Integration (#3)

- Change order to have Language Definition before Langium Integration
- Explain how languages can be defined in Langium
- Explain Model Hub-Langium integration library and layer

CamilleLetavernier committed:
doc: Editing Domain Section (#4)

eneufeld committed:
remove references to new code base and cleanup

CamilleLetavernier committed:
doc: improve Model Hub section (#2)

* describe model hub section
* add content about reference resolution and cross references

---------
Authored-by: Camille Letavernier <cletavernier@eclipsesource.com>
Co-authored-by: Martin Fleck <mfleck@eclipsesource.com>

planger commented on content/_index/coffeeeditoroverview/index.md:
```diff
@@ -3,11 +3,12 @@ fragment = "content"
 </p>
-<img src="images/coffeeeditordemo.gif" alt="Coffee Editor Overview" style="display: block; margin: auto;"/>
\ No newline at end of file
+<img src="images/coffeeeditordemo.gif" alt="Coffee Editor Overview" style="display: block; margin: auto;"/>
```
Can we already provide an updated GIF or preferably a video that shows the NG. If possible in dark mode?

eneufeld replied:
it looks the same ;-).

planger commented on content/documentation/additionals/content.md:
```diff
@@ -0,0 +1,27 @@
+
+## Videos
+
+* EclipseCon 2021: <a target="_blank" href="https://www.youtube.com/watch?v=mEr-v8t2ZQ8">Model validation, diffing and more with EMF.cloud</a>
```
```suggestion
* EclipseCon 2023: <a target="_blank" href="https://www.youtube.com/watch?v=U0TRt3aukTc">Building cloud-native (modeling) tools</a>
* EclipseCon 2021: <a target="_blank" href="https://www.youtube.com/watch?v=mEr-v8t2ZQ8">Model validation, diffing and more with EMF.cloud</a>
```

planger commented on content/documentation/modelingLanguage/content.md:
```diff
@@ -0,0 +1,138 @@
+  sticky = true
++++
+
+In Model Hub, languages can be easily defined using Langium. 
```
```suggestion
In Model Hub, languages can be easily defined using Eclipse Langium. 
```

planger commented on content/documentation/langium/content.md:
```diff
@@ -0,0 +1,99 @@
+Furthermore, we need to ensure that anytime a model is updated on the Langium side we properly update the model on the Model Hub side.
+We achieve that by installing a listener on the Langium side and using the Model Manager from the Model Hub to execute a PATCH command that updates the model in the Model Hub.
+
+For the Coffe Model, the persistence contribution may look something like this:
```
```suggestion
For the Coffee Model, the persistence contribution may look something like this:
```

planger reviewed (changes requested):
Very nice, thank you!

I think we should update the links here too:
![image](https://github.com/eclipse-emfcloud/emfcloud-website-source/assets/588090/99fd4bf4-23ef-434b-90b7-351641234b47)

1. In Browse the sources, we should either mention that the sources for NG are available soon and/or explicitly say that the sources linked here are pointing to the Java-based "product line". I prefer the first to avoid confusion, as this section is actually about NG.
2. In the Lean more section, I'd just point to [Maximilian's talk at EclipseCon](https://www.youtube.com/watch?v=U0TRt3aukTc) and the docs, but not to the blog post about the Java-based product line.

eneufeld committed:
incorporate review comments

eneufeld committed:
rename `EMF.cloud` to `EMF Cloud`

planger reviewed (approved):
Thanks, looks good to me! :+1:

The only thing I noticed, is that there is a documentation page that still has todos on it:
https://deploy-preview-45--emfcloud.netlify.app/documentation/diagrameditor/

Imho this can be also done after merging, but we should not wait too long and prioritize completing this page to give the full picture.

Thank you!

eneufeld commented:
Nina promised me an update tomorrow (21.12) morning

ndoschek committed:
doc: add diagram editor documentation

- add documentation for diagram editors
- update outdated animated diagram gif (from outdated ecore glsp editor)
